File: packages/room-server/src/node/entities/node.entity.ts
import { IFormProps } from '@apitable/core';
import { BaseEntity } from 'shared/entities/base.entity';
import { Column, Entity } from 'typeorm';

/**
 * Workbench-Node
 */
@Entity('node')
export class NodeEntity extends BaseEntity {
  @Column({
    name: 'type',
    nullable: false,
    comment: 'node type(0: root node; 1: folder; 2: datasheet; 3: form)',
    width: 2,
    type: 'tinyint',
  })
  type!: number;

  @Column({
    name: 'space_id',
    nullable: false,
    comment: 'space ID(related#space#space_id)',
    length: 50,
  })
  spaceId!: string;

  @Column({
    name: 'unit_id',
    type: 'bigint',
    nullable: false,
    comment: 'unit ID',
    default: '0',
  })
  unitId!: string;

  @Column({
    name: 'parent_id',
    nullable: false,
    comment: 'parent node ID',
    length: 50,
  })
  parentId!: string;

  @Column({
    name: 'pre_node_id',
    nullable: true,
    comment: 'previous node ID',
    length: 50,
  })
  preNodeId?: string;

  @Column({
    name: 'node_id',
    nullable: false,
    unique: true,
    comment: 'node Id(related#node#node_id)',
    length: 50,
  })
  nodeId!: string;

  @Column({
    name: 'node_name',
    nullable: true,
    comment: 'node name',
    length: 255,
  })
  nodeName!: string;

  @Column({
    name: 'icon',
    nullable: true,
    comment: 'node icon',
    length: 100,
  })
  icon?: string;

  @Column({
    name: 'cover',
    nullable: true,
    comment: 'Cover TOKEN',
    length: 255,
  })
  cover?: string;

  @Column({
    name: 'is_template',
    comment: 'is it a template(0: no, 1: yes)',
    unsigned: true,
    default: () => false,
  })
  isTemplate!: boolean;

  @Column('json', {
    name: 'extra',
    nullable: true,
    comment: 'other information',
  })
  extra?: IFormProps;

  @Column({
    name: 'deleted_path',
    nullable: true,
    comment: 'The path when deletes',
    length: 255,
  })
  deletedPath?: string;

  @Column({
    name: 'is_rubbish',
    comment: 'recycling station mark (0: no, 1: yes)',
    unsigned: true,
    default: () => false,
  })
  isRubbish!: boolean;

  @Column({
    name: 'is_banned',
    comment: 'banning sign (0: No, 1: Yes)',
    unsigned: true,
    default: () => false,
  })
  isBanned!: boolean;
}

File: backend-server/application/src/main/java/com/apitable/workspace/entity/NodeEntity.java


package com.apitable.workspace.entity;

import com.baomidou.mybatisplus.annotation.FieldFill;
import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableLogic;
import com.baomidou.mybatisplus.annotation.TableName;
import java.io.Serializable;
import java.time.LocalDateTime;
import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.NoArgsConstructor;
import lombok.experimental.Accessors;

/**
 * <p>
 * Workbench -  Node Table.
 * </p>
 *
 * @author Mybatis Generator Tool
 */
@Data
@Builder(toBuilder = true)
@NoArgsConstructor
@AllArgsConstructor
@Accessors(chain = true)
@EqualsAndHashCode
@TableName(keepGlobalPrefix = true, value = "node")
public class NodeEntity implements Serializable {

    private static final long serialVersionUID = 1L;

    /**
     * Primary key.
     */
    @TableId(value = "id", type = IdType.ASSIGN_ID)
    private Long id;

    /**
     * Space ID(link#xxxx_space#space_id).
     */
    private String spaceId;

    /**
     * unit id(link#xxxx_unit#id).
     */
    private Long unitId;

    /**
     * Parent Node Id.
     */
    private String parentId;

    /**
     * ID of the previous node under the same level.
     */
    private String preNodeId;

    /**
     * Custom Node ID.
     */
    private String nodeId;

    /**
     * Node Name.
     */
    private String nodeName;

    /**
     * Node Icon.
     */
    private String icon;

    /**
     * Type (0:Root node,1:Folder,2:Datasheet).
     */
    private Integer type;

    /**
     * Cover Draw TOKEN.
     */
    private String cover;

    /**
     * Is Template (0: No, 1: Yes).
     */
    private Boolean isTemplate;

    /**
     * Other information.
     */
    private String extra;

    /**
     * Creator.
     */
    private Long creator;

    /**
     * deleted path.
     */
    private String deletedPath;

    /**
     * Delete tag(0:No,1:Yes).
     */
    @TableLogic
    private Boolean isDeleted;

    /**
     * Recycle Bin Tag (0: No, 1: Yes).
     */
    private Boolean isRubbish;

    /**
     * Banned or not (0: No, 1: Yes).
     */
    private Boolean isBanned;

    /**
     * Creator.
     */
    @TableField(fill = FieldFill.INSERT)
    private Long createdBy;

    /**
     * Last Update By.
     */
    @TableField(fill = FieldFill.INSERT_UPDATE)
    private Long updatedBy;

    /**
     * Create Time.
     */
    private LocalDateTime createdAt;

    /**
     * Update Time.
     */
    private LocalDateTime updatedAt;

}
